dipy/align/imaffine.py

```diff
@@ -0,0 +1,715 @@
+                                        self.moving_vals,
+                                        self.samples[..., :self.dim],
+                                        mgrad)
+
+        # Evaluate the mutual information and its gradient
+        # The results are in self.metric_val and self.metric_grad
+        # ready to be returned from 'distance' and 'gradient'
+        self.update_mi_metric(update_gradient)
+
+    def distance(self, xopt):
+        r""" Numeric value of the negative Mutual Information
+
+        We need to change the sign so we can use standard minimization
+        algorithms.
+
+        Parameters
+        ----------
+        xopt : array, shape (n,)
+            the parameter vector of the transform currently used by the metric
+            (the transform name is provided when self.setup is called), n is
+            the number of parameters of the transform
+        """
+        if self.samples is None:
+            self._update_dense(xopt, False)
+        else:
+            self._update_sparse(xopt, False)
+        return -1 * self.metric_val
+
+    def gradient(self, xopt):
+        r""" Numeric value of the metric's gradient at the given parameters
+
+        Parameters
+        ----------
+        xopt : array, shape (n,)
+            the parameter vector of the transform currently used by the metric
+            (the transform name is provided when self.setup is called), n is
+            the number of parameters of the transform
+        """
+        if self.samples is None:
+            self._update_dense(xopt, True)
+        else:
+            self._update_dense(xopt, True)
+        return self.metric_grad * (-1)
+
+    def value_and_gradient(self, xopt):
+        r""" Numeric value of the metric and its gradient at given parameters
+
+        Parameters
+        ----------
+        xopt : array, shape (n,)
+            the parameter vector of the transform currently used by the metric
+            (the transform name is provided when self.setup is called), n is
+            the number of parameters of the transform
+        """
+        if self.samples is None:
+            self._update_dense(xopt, True)
+        else:
+            self._update_sparse(xopt, True)
+        return -1 * self.metric_val, self.metric_grad * (-1)
+
+
+class AffineRegistration(object):
+    def __init__(self,
+                 metric=None,
+                 level_iters=None,
+                 opt_tol=1e-5,
+                 sigmas=None,
+                 factors=None,
+                 method='L-BFGS-B',
+                 ss_sigma_factor=None,
+                 options=None):
+        r""" Initializes an instance of the AffineRegistration class
+
+        Parameters
+        ----------
+        metric : object
+            an instance of a metric
+        level_iters : list
+            the number of iterations at each level of the Gaussian pyramid.
+            level_iters[0] corresponds to the finest level, level_iters[n-1]
```

Comment:
Add single quotes around `level_iters[0]` and `level_iters[n-1]`. Also, that can probably be `level_iters[-1]`, no?